subprojects/common/elastic/src/main/kotlin/com/avito/android/elastic/DateFormatCheckerImpl.kt

```diff
@@ -0,0 +1,21 @@
+package com.avito.android.elastic
+
+import java.util.Locale
+
+internal class DateFormatCheckerImpl : DateFormatChecker {
+
+    private val datePattern = Regex("[0-9]{4}-[0-9]{2}-[0-9]{2}")
+
+    private val dateCheckCache = mutableSetOf<String>()
+
+    override fun check(formattedDate: String) {
+        if (!dateCheckCache.contains(formattedDate)) {
```

Comment:
It seems to me like premature optimization.

Why do you think that this regex will be a problem?
Stored dates can bloat memory.

Reply:
this date must have a `yyyy-MM-dd` format
i see some invalid format evidence:

```
2021.1.20
2021.01.0020
2021.0001.20
2021.0001.0020
```

Don't know it could happen, so this check is to find bad locale of something

i don't want to calcualte regex match on every log call, so something like `2020.01.20` will be in cache for whole build